src/library/autodj/autodjprocessor.cpp

```diff
@@ -644,6 +646,8 @@ void AutoDJProcessor::playerPositionChanged(DeckAttributes* pAttributes,
     if (thisPlayPosition >= thisDeck.fadeBeginPos && thisDeck.isFromDeck) {
         if (m_eState == ADJ_IDLE) {
             if (thisDeckPlaying || thisPlayPosition >= 1.0) {
+                // cache this before calculating the new transition in otherDeck.play();
+                bool hasFadeTransition = thisDeck.fadeBeginPos < thisDeck.fadeEndPos;
```

Comment:
When could this be false? It is not obvious. At first glance it seems like this should be a debug assert.

Reply:
This is always false, if you use a negative or zero transition time.
I will add a comment.